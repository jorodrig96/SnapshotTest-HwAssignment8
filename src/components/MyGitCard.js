import Card from 'react-bootstrap/Card';
import { useState, useEffect } from 'react';

export default function MyGitCard() {
    
const [gitHubName, setGitHubName] = useState('')                     

useEffect(() => {
    fetch('https://api.github.com/users/jorodrig96')
    .then(res => res.json())
    .then(data => {
        setGitHubName(data.name)
    })
    }, []);
    
  return (
    <Card className="bg-dark text-white">
      <Card.Img className='imgCard' src="https://steamuserimages-a.akamaihd.net/ugc/960856720981536415/90FC611EA20B1797EABD82F92508DBB2DAFBBB0E/?imw=5000&imh=5000&ima=fit&impolicy=Letterbox&imcolor=%23000000&letterbox=false" alt="Card image" />
      <Card.ImgOverlay>
        <Card.Title>MockFetch-Assignment8</Card.Title>
        <Card.Text>
          Hello, my name is {gitHubName}, & I hope to be doing this as a carreer soon. 
          <p>I'm excited to start this new journey.</p>
          <h6>Yours truly, {gitHubName} :)</h6>
        </Card.Text>
      </Card.ImgOverlay>
    </Card>
  );
};