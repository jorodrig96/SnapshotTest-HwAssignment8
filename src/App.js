import './App.css';
import MyGitCard from './components/MyGitCard';


function App() {
  return (
    <div className="App">
      <MyGitCard />
    </div>
    );
}

export default App;
