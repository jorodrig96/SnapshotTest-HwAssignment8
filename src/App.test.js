import { render, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import MyGitCard from './components/MyGitCard';


describe('rendering tests', () => {
  test('rendering a test', () => {
    const tree = renderer.create(<App />).toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot()
  })
});

