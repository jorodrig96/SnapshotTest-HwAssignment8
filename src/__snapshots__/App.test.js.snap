// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering tests rendering a test 1`] = `
<div
  className="App"
>
  <div
    className="bg-dark text-white card"
  >
    <img
      alt="Card image"
      className="card-img imgCard"
      src="https://steamuserimages-a.akamaihd.net/ugc/960856720981536415/90FC611EA20B1797EABD82F92508DBB2DAFBBB0E/?imw=5000&imh=5000&ima=fit&impolicy=Letterbox&imcolor=%23000000&letterbox=false"
    />
    <div
      className="card-img-overlay"
    >
      <div
        className="card-title h5"
      >
        MockFetch-Assignment8
      </div>
      <p
        className="card-text"
      >
        Hello, my name is 
        , & I hope to be doing this as a carreer soon.
        <p>
          I'm excited to start this new journey.
        </p>
        <h6>
          Yours truly, 
           :)
        </h6>
      </p>
    </div>
  </div>
</div>
`;
